xtask/src/ast_src.rs

```diff
@@ -533,70 +1609,359 @@ pub(crate) const AST_SRC: AstSrc = AstSrc {
             const_params: [ConstParam],
             T![>]
         }
+
+        /// Single type parameter **declaration**.
+        ///
+        /// ```
+        /// fn foo<❰ K ❱, ❰ I ❱, ❰ E: Debug ❱, ❰ V = DefaultType ❱>() {}
+        /// ```
+        ///
+        /// [Reference](https://doc.rust-lang.org/reference/items/generics.html)
         struct TypeParam: NameOwner, AttrsOwner, TypeBoundsOwner {
             T![=],
             default_type: TypeRef,
         }
+
+        /// Const generic parameter **declaration**.
+        /// ```
+        /// fn foo<T, U, ❰ const BAR: usize ❱, ❰ const BAZ: bool ❱>() {}
+        /// ```
+        ///
+        /// [RFC](https://github.com/rust-lang/rfcs/blob/master/text/2000-const-generics.md#declaring-a-const-parameter)
         struct ConstParam: NameOwner, AttrsOwner, TypeAscriptionOwner {
             T![=],
             default_val: Expr,
         }
+
+        /// Lifetime parameter **declaration**.
+        ///
+        /// ```
+        /// fn foo<❰ 'a ❱, ❰ 'b ❱, V, G, D>(bar: &'a str, baz: &'b mut str) {}
+        /// ```
+        ///
+        /// [Reference](https://doc.rust-lang.org/reference/items/generics.html)
         struct LifetimeParam: AttrsOwner { T![lifetime] }
-        struct TypeBound { T![lifetime], /* Question,  */ T![const], /* Question,  */ TypeRef}
+
+        // TODO: better clarify where is the colon token and what `const` pertains to.
+        // TODO: add example with `const`
```

Comment:
https://github.com/rust-lang/rust/blob/6af388b25050bca26710be7e4030e17bf6d8d2f7/src/test/ui/rfc-2632-const-trait-impl/const-trait-bound-opt-out/in-trait-bounds.rs AFAIK

Reply:
Thanks, added const example in eedf11ae8897083ba4d5f64e6bce38f4e19cd8e2